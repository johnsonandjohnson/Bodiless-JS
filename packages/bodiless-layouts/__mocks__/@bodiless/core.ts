const editContext = {
  activate: jest.fn(),
  isEdit: true,
};

export const useEditContext = jest.fn(() => editContext);

const activateOnEffect = {
  setId: jest.fn(),
};
export const useActivateOnEffect = () => activateOnEffect;

export const useNode = jest.fn();

const contextMenuFormInner = jest.fn();
export const contextMenuForm = jest.fn(() => contextMenuFormInner);
