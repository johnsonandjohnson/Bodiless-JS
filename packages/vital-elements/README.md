# `@bodiless/vital-elements`

For documentation on the Vital Elements package, please see:

- [Vital Elements | BodilessJS Docs](https://johnsonandjohnson.github.io/Bodiless-JS/#/VitalDesignSystem/Components/VitalElements/)
- [`doc/` Directory](./doc)
