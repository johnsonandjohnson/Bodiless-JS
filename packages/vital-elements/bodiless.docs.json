{
  "VitalDesignSystem": {
    "README.md": "./doc/README.md",
    "Guides": {
      "README.md": "./doc/Guides.md",
      "RepositoryStructure.md": "./doc/RepositoryStructure.md",
      "ComponentTemplate.md": "./doc/ComponentTemplate.md",
      "Tokens": {
        "README.md": "./doc/Tokens.md",
        "TokenDomains.md": "./doc/TokenDomains.md"
      },
      "ExtendingAndComposingTokens.md": "./doc/ExtendingAndComposing.md",
      "ShadowingTokens.md": "",
      "TailwindGuide.md": "./doc/TailwindGuide.md",
      "SiteTypography.md": "./doc/SiteTypography.md",
      "StaticReplacement.md": "./doc/StaticReplacement.md"
    },
    "Components": {
      "README.md": "./doc/Components.md",
      "VitalElements": {
        "README.md": "./doc/VitalElements.md"
      },
      "assets": {
        "PXtoREMTemplate.xlsx": "./doc/assets/PXtoREMTemplate.xlsx"
      }
    }
  }
}
