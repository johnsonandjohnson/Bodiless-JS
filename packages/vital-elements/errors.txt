Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Secondary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Tertiary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Light Theme/Idle
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Form Field/Text/Caption
 . Component color alias "(Typography/Body/Small" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Success
 . Component color alias "(Color/Signal/Success/Success" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Input
 . Component color alias "(Typography/Body/Regular" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Label
 . Component color alias "(Typography/Eyebrow/Eyebrow" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Link
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Hover
 . Component color alias "(Semantic/Color/Interactive/Default" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Active
 . Component color alias "(Semantic/Color/Interactive/Hover" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Bottom
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Top
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Background/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Background/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Accordion/Dropdown/Border/Light Theme/Border
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Light Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Dark Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
