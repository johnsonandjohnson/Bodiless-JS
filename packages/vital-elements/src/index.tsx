/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export * from './components/AccordionElement';
export * from './components/ArticleCardElement';
export * from './components/BreadcrumbElement';
export * from './components/ButtonElement';
export * from './components/CategoryCardElement';
export * from './components/Color';
export * from './components/DividerElement';
export * from './components/FilterElement';
export * from './components/FloatingButtonElement';
export * from './components/FontSize';
export * from './components/FooterElement';
export * from './components/HeroBannerElement';
export * from './components/ImageCarouselElement';
export * from './components/InputElement';
export * from './components/KnapsackDemoElement';
export * from './components/LinkElement';
export * from './components/ModalElement';
export * from './components/NavigationElement';
export * from './components/PDPProductOverviewElement';
export * from './components/ProductCardElement';
export * from './components/ScrollIndicatorElement';
export * from './components/Spacing';
export * from './components/TextDecoration';
export * from './components/KnapsackDemoElement';
export { default as vitalScrollIndicatorElement } from './generated/vitalScrollIndicatorElement';
export * from './components/Typography';
export * from './util';
