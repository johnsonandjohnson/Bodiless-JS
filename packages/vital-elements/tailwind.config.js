/* eslint-disable import/no-dynamic-require, global-require */
/**
 * Copyright Â© 2021 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { getPackageTailwindConfig } from '@bodiless/fclasses';

const tailwindcssDir = require('tailwindcss-dir')();
const designTokens = require('./assets/design-tokens.nested.json');

// Parse the JSON Tokens data
const parsedDesignTokens = JSON.parse(JSON.stringify(designTokens));

/**
 * The `design-tokens.nested.json` file should only contain all Core tokens.
 * These core tokens are then used to extend the corresponding Tailwind config sections.
 *
 * @TODO: The `border.style` tokens do not have a corresponding key in the Tailwind config.
 * `borderStyle` is only a valid key for the TW variants and plugins. The default TW values for
 * borderStyle match the provided tokens.
 *
 * @TODO: `typography` tokens include `text-decoration` and `text-transform` tokens which are
 * not intended to be extended or modified in the Tailwind config and do not have corresponding
 * keys in the Tailwind config. The default TW config covers these tokens and provides the same
 * Default classes like `capitalize` or `uppercase`.
 *
 * @TODO: Also note that currently the `design-tokens.nested.json` is manually adjusted to
 * the more favourable format to resolve certain issues with resolving Figma Plugin aliases.
 * Thehere is a script being worked on to format tokens automatically.
 *
 * @TODO: The `border-width` and `border-style` tokens in JSON are combined under the same key.
 * Tailwind treats it as two separate styles with different config key. In manually updated JSON
 * it is split into two different token "section" like `border.width` and `boder.style`
 * to avoid TW config issues.
 */
const {
  // colors,
  // spacing,
  // typography,
  opacity,
  // border,
} = parsedDesignTokens;

const resolver = (pkgName) => require.resolve(pkgName);

const twConfig = {
  content: [
    './lib/**/!(*.d).{ts,js,jsx,tsx}',
  ],
  theme: {
    fontSize: {
      xs: ['0.75rem', '1.125rem'],
      sm: ['0.875rem', '1.25rem'],
      base: ['1rem', '1.375rem'],
      lg: ['1.75rem', '1.875rem'],
      xl: ['1.625rem', '2rem'],
      '2xl': ['2rem', '2.25rem'],
      '3xl': ['2.563rem', '2.813rem'],
      'm-xs': ['0.688rem', '0.938rem'],
      'm-sm': ['0.75rem', '1rem'],
      'm-base': ['0.875rem', '1.125rem'],
      'm-lg': ['1.125rem', '1.375rem'],
      'm-xl': ['1.438rem', '1.625rem'],
      'm-2xl': ['1.813rem', '2.125rem'],
      'm-3xl': ['2.25rem', '2.5rem'],
    },
    colors: {
      transparent: 'transparent',
      'vital-primary': {
        brand: '#CA081B',
        'card-bg': '#ffffff',
        'page-bg': '#F4F4F4',
        interactive: '#019881',
        'interactive-active': '#000341',
        divider: '#D8D8D8',
        'body-copy': '#63666A',
        'header-copy': '#212121',
        'footer-copy': '#FFFFFF',
        'button-selected': '#015B4D',
        'button-disabled': '#999999',
      },
      'vital-secondary': {
        eyebrow: '#CC0099',
        'footer-bg': '#2B2B33',
        interactive: '#018571',
        table: '#E5E7EB',
        'table-interactive': '#f3f4f6',
        separator: '#666666',
        search: '#888888',
        'button-selected': '#B5E1DA',
        'button-hovered': '#E1F3F0',
        'button-disabled': '#999999',
        'border-button-selected': '#015B4D',
      },
      white: '#FFFFFF',
    },
    extend: {
      /**
       * Vital 2.0 Tokens coming from Figma plugin.
       */
      colors: {
        // ...colors,
        ColorButtonPrimaryTextLightThemeText: '#ffffff',
        ColorInteractiveLightThemeIdle: '#019881',
        ColorInteractiveLightThemeHover: '#017A67',
        ColorInteractiveLightThemeFocus: '#017A67',
        ColorInteractiveLightThemePressed: '#015B4D',
        ColorInteractiveLightThemeDisabled: '#999999',
      },
      fontFamily: {
        DMSans: ['DM Sans', 'sans-serif'],
        // ...typography['font-family'],
        1: 'DM Sans',
        2: 'DM Sans',
      },
      fontWeights: {
        // ...typography.font,
      },
      fontSize: {
        '14px': '14px',
        '16px': '16px',
      },
      lineHeight: {
        // ...typography.leading,
        5: '120%',
        6: '150%'
      },
      textSizes: {
        // ...typography.text,
      },
      spacing: {
        // ...spacing,
        '0px': '0px',
        '1px': '1px',
        '2px': '2px',
        '3px': '3px',
        '4px': '4px',
        '5px': '4px',
        '6px': '4px',
        '7px': '4px',
        '8px': '4px',
        '9px': '4px',
        '10px': '4px',
        '12px': '4px',
        '14px': '4px',
        '16px': '4px',
        '18px': '4px',
        '20px': '4px',
        '24px': '4px',
        '32px': '4px',
        '36px': '4px',
        '40px': '4px',
        '48px': '4px',
        '64px': '4px',
      },
      borderWidth: {
        // ...border.width,
        '1px': '1px',
      },
      borderRadius: {
        // ...border.rounded,
        0: '0px',
        2: '2px',
        4: '4px',
        6: '6px',
        8: '8px',
        10: '10px',
        12: '12px',
        20: '20px',
        150: '150px',
        600: '600px',
        pill: '9999px',
      },
      opacity: {
        // ...opacity,
      },
    },
  },
  plugins: [
    tailwindcssDir,
  ],
};

module.exports = getPackageTailwindConfig({
  twConfig,
  resolver,
});
