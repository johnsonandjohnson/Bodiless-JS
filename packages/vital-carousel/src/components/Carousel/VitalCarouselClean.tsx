import React, { FC, Fragment } from 'react';
import { asVitalTokenSpec } from '@bodiless/vital-elements';
import { designable, Div, Ul } from '@bodiless/fclasses';
import { Helmet } from 'react-helmet';
import type { DesignableComponentsProps } from '@bodiless/fclasses';
import type { VitalCarouselComponents } from './types';
import { carouselScript } from './utils/carousel';
import vitalCarouselTokens from '../CarouselTokens';

type VitalCarouselBaseProps = DesignableComponentsProps<VitalCarouselComponents>;

const carouselActive = vitalCarouselTokens.ScrollIndicatorInactive;
const carouselInActive = vitalCarouselTokens.ScrollIndicatorActive;

const cssString = `:root { --carouselActive: ${carouselActive}; --carouselInActive: ${carouselInActive}; }`;

const CarouselScript = () => (
  <Helmet>
    <script type="text/javascript">
      {carouselScript}
    </script>
    <style>
      {cssString}
    </style>
  </Helmet>
);

/**
 * The starting components for each slot.
 */
const vitalCarouselComponents: VitalCarouselComponents = {
  Wrapper: Div,
  Script: CarouselScript,
  SliderWrapper: Div,
  Slider: Ul,
  Slide: Div,
  ControlsWrapper: Div,
  Indicator: Fragment,
};

const VitalCarouselBase: FC<VitalCarouselBaseProps> = ({ components: C, ...rest }) => (
  <C.Wrapper {...rest}>
    <C.Script />
    <C.SliderWrapper>
      <C.Slider />
    </C.SliderWrapper>
    <C.ControlsWrapper>
      <C.Indicator />
    </C.ControlsWrapper>
  </C.Wrapper>
);

const VitalCarouselClean = designable(vitalCarouselComponents, 'VitalCarousel')(VitalCarouselBase);

/**
 * A token creator that respects the VitalCarousel slots.
 *
 * @category Token Collection
 */
export const asVitalCarouselToken = asVitalTokenSpec<VitalCarouselComponents>();

export default VitalCarouselClean;
