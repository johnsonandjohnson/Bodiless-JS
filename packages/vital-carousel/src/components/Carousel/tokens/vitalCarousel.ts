import {
  withChild,
} from '@bodiless/core';
import { withNodeKey } from '@bodiless/data';
import {
  Img, Div, as, on, replaceWith, stylable, flowHoc,
  withDesign, Fragment,
} from '@bodiless/fclasses';
import { vitalImage } from '@bodiless/vital-image';
import { vitalCard, CardClean } from '@bodiless/vital-card';
import { asBodilessList } from '@bodiless/components';
// import vitalCarouselTokens from '../../CarouselTokens';
import { asVitalCarouselToken } from '../VitalCarouselClean';
import type { VitalCarousel } from '../types';
import '../../../../assets/scroll-snap-slider.css';

import { CAROUSEL_NODE_KEY } from '../utils/constants';
import CarouselDot from '../utils/CarouselDot';
import CarouselSlide from '../utils/CarouselSlide';
import CarouselThumb from '../utils/CarouselThumb';

const Default = asVitalCarouselToken({
  Components: {
    Slider: flowHoc(
      asBodilessList(CAROUSEL_NODE_KEY, undefined, () => ({ groupLabel: 'Slide' })),
      withDesign({
        Item: replaceWith(stylable(CarouselSlide)),
      }),
    ),
  },
  Behavior: {
    Wrapper: 'slider',
    SliderWrapper: '',
    Slider: as(
      'flex flex-nowrap justify-normal overflow-x-auto',
      'scroll-smooth snap-always snap-x',
      'scroll-snap-slider -simple',
      withDesign({
        Item: as(
          'scroll-snap-slide',
          'items-center flex flex-col justify-center max-w-none',
          'shrink-0 grow-0',
          'snap-start',
        ),
      }),
    ),
  },
});

const WithControls = asVitalCarouselToken({
  Components: {
    Dots: as(
      replaceWith(Div),
      asBodilessList(CAROUSEL_NODE_KEY, undefined, () => ({ groupLabel: 'Slide' })),
      'controls indicators',
      'align-center flex-row justify-center opacity-100',
    ),
  },
});

const WithCarouselDots = asVitalCarouselToken(
  WithControls,
  {
    Components: {
      Dots: withDesign({
        Item: as(
          replaceWith(
            withChild(
              stylable(CarouselDot), 'DotIndicator',
            )(Fragment),
          ),
        ),
      }),
    },
    Theme: {
      Slider: as(
        '-multi space-x-1',
        withDesign({
          Item: 'w-5/6 md:w-1/3 lg:w-1/4',
        }),
      ),
      ControlsWrapper: 'flex pt-2',
      Dots: 'flex items-center dots -simple space-x-2 lg:hidden', // Spacing-8
    },
  }
);

const WithThumbnail = asVitalCarouselToken(
  WithControls,
  {
    Components: {
      Dots: withDesign({
        Item: as(
          replaceWith(
            withChild(
              stylable(CarouselThumb), 'ThumbIndicator',
            )(Fragment),
          ),
        ),
      }),
    },
    Theme: {
      Slider: as(
        withDesign({
          Item: 'w-full',
        }),
      ),
      Dots: withDesign({
        Item: withDesign({
          Dot: as(
            'flex items-center indicators -simple',
          ),
        }),
      }),
    },
    Layout: {
      ControlsWrapper: 'flex justify-left',
      Dots: 'flex items-center space-x-1',
    },
    Spacing: {
      ControlsWrapper: 'pt-6',
    },
  }
);

// const asAccessibleCarouselButton = flowHoc(
//   addProps({
//     role: 'button',
//   }),
// );

// const withAriaSelectedCarouselItem = flowHoc(
//   addPropsIf(useIsCarouselItemActive)({
//     'aria-selected': true,
//     'aria-hidden': false,
//   }),
//   addPropsIf(negate(useIsCarouselItemActive))({
//     'aria-selected': false,
//     'aria-hidden': true,
//   }),
// );

// const asAccessibleCarousel = asVitalCarouselToken({
//   A11y: {
//     Slider: flowHoc(
//       addProps({
//         tabIndex: 'auto',
//       }),
//       withDesign({
//         Item: flowHoc(
//           withCarouselItemTabIndex,
//           withAriaSelectedCarouselItem,
//         ),
//       }),
//     ),
//     ButtonBack: asAccessibleCarouselButton,
//     ButtonNext: asAccessibleCarouselButton,
//     Dots: withDesign({
//       Item: flowHoc(
//         withAriaSelectedCarouselItem,
//         addProps({
//           'aria-hidden': false,
//           role: 'presentation',
//         }),
//         withDesign({
//           Dot: asAccessibleCarouselButton,
//         }),
//       ),
//     }),
//     ButtonPlay: asAccessibleCarouselButton,
//   }
// });

const WithImageSlide = asVitalCarouselToken({
  Components: {
    Slider: withDesign({
      Title: on(Img)(
        vitalImage.Default,
        withNodeKey('image'),
      )
    }),
  }
});

const WithCardSlide = asVitalCarouselToken({
  Components: {
    Slider: withDesign({
      Title: on(CardClean)(
        vitalCard.Product,
        withNodeKey('card'),
      )
    }),
  },
});

// These are not ideal but allows to swap between tokens on responsive
const MobileOnly = asVitalCarouselToken({
  Layout: {
    Wrapper: 'md:hidden'
  },
});

const TabletOnly = asVitalCarouselToken({
  Layout: {
    Wrapper: 'hidden md:block lg:hidden'
  },
});

const TabletDesktopOnly = asVitalCarouselToken({
  Layout: {
    Wrapper: 'hidden md:block'
  },
});

const DesktopOnly = asVitalCarouselToken({
  Layout: {
    Wrapper: 'hidden lg:block'
  },
});

/**
 * Tokens for VitalCarouselClean
 *
 * @category Token Collection
 * @see [[VitalCarousel]]
 * @see [[VitalCarouselClean]]
 */
const vitalCarousel: VitalCarousel = {
  Default,
  // WithNavigationButtons,
  WithControls,
  WithCarouselDots,
  WithThumbnail,
  // asAccessibleCarousel,
  WithImageSlide,
  WithCardSlide,
  MobileOnly,
  TabletOnly,
  TabletDesktopOnly,
  DesktopOnly,
};

export default vitalCarousel;
