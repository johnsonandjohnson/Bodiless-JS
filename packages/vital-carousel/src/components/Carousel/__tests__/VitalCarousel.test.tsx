import React from 'react';
import { testTokens } from '@bodiless/vital-elements';
import VitalCarouselClean from '../VitalCarouselClean';
import vitalCarousel from '../tokens';

// eslint-disable-next-line jest/valid-describe
describe('VitalCarousel Tokens', testTokens(VitalCarouselClean, vitalCarousel));
