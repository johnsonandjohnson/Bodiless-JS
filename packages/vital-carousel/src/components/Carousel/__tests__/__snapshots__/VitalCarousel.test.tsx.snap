// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VitalCarousel Tokens Matches snapshot for Default 1`] = `
<div
  aria-label="carousel"
  className="slider"
  role="region"
>
  <div>
    <ul
      className="scrollbar scroll-smooth snap-always snap-x pb-20px md:pb-24px flex flex-nowrap justify-normal overflow-x-auto"
    >
      <li
        className="snap-start scroll-snap-slide shrink-0 grow-0 items-center flex flex-col justify-center max-w-none"
        data-index={0}
        name="slide_switch"
        role="list"
      />
    </ul>
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for DesktopOnly 1`] = `
<div
  className="hidden lg:flex"
>
  <div>
    <ul />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for MobileOnly 1`] = `
<div
  className="md:hidden"
>
  <div>
    <ul />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for MobileTabletOnly 1`] = `
<div
  className="lg:hidden"
>
  <div>
    <ul />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for TabletDesktopOnly 1`] = `
<div
  className="hidden md:flex"
>
  <div>
    <ul />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for TabletOnly 1`] = `
<div
  className="hidden md:flex lg:hidden"
>
  <div>
    <ul />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithCardSlide 1`] = `
<div>
  <div>
    <ul
      design={
        Object {
          "Title": [Function],
        }
      }
    />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithCarouselDots 1`] = `
<div>
  <div>
    <ul
      className="scroll-snap-dot-slider scrollbar-hide lg:scrollbar-default"
      design={
        Object {
          "Item": [Function],
        }
      }
    />
  </div>
  <div
    className="pt-2 flex"
  >
    <div
      aria-label="Navigation Controls"
      className="ps-2 flex items-center dots -simple lg:hidden align-center justify-center opacity-100 controls indicators list-none"
      role="list"
    >
      <div
        data-no-hydrate={true}
        id="c89d2b4890105bf8d2e48cb608de0a29"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <div
          className="indicator pe-2"
          data-index={0}
          tabIndex="-1"
        >
          <span
            className="bg-kenvue-neutrals-light-grey rounded-full border-none dot block w-[8px] h-[8px]"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithCarouselDotsAllViewports 1`] = `
<div>
  <div>
    <ul
      className="w-screen md:w-full"
      design={
        Object {
          "Item": [Function],
        }
      }
    />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithCarouselDotsMobileTablet 1`] = `
<div>
  <div>
    <ul
      className="w-screen md:w-full"
      design={
        Object {
          "Item": [Function],
        }
      }
    />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithControls 1`] = `
<div>
  <div>
    <ul />
  </div>
  <div>
    <div
      aria-label="Navigation Controls"
      className="align-center justify-center opacity-100 controls indicators list-none"
      role="list"
    >
      <li />
    </div>
  </div>
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithHorizontalThumbs 1`] = `
<div
  className="lg:flex-col"
>
  <div>
    <ul />
  </div>
  <div
    className="pt-6 flex-row"
  >
    <div
      className="p-8px flex flex-col justify-between thumbcontrols"
    >
      <div
        className="text-vital-primary-interactive hover:opacity-70 active:text-vital-primary-interactive-active arrow -prev"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="16"
            id="mask0_3721_63618"
            maskUnits="userSpaceOnUse"
            style={
              Object {
                "maskType": "alpha",
              }
            }
            width="24"
            x="0"
            y="4"
          >
            <path
              d="M8.04428 19L1.04878 11.9954L8.05339 5"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
            <path
              d="M1.47635 12.0409L22.8811 12.0409"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeWidth="2"
            />
          </mask>
          <g
            mask="url(#mask0_3721_63618)"
            style={
              Object {
                "maskType": "alpha",
              }
            }
          >
            <rect
              fill="currentColor"
              height="24"
              transform="rotate(-180 23.9299 24)"
              width="24"
              x="23.9299"
              y="24"
            />
          </g>
        </svg>
      </div>
      <div
        className="text-vital-primary-interactive hover:opacity-70 active:text-vital-primary-interactive-active arrow -next"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="16"
            id="mask0_3721_63619"
            maskUnits="userSpaceOnUse"
            style={
              Object {
                "maskType": "alpha",
              }
            }
            width="24"
            x="0"
            y="4"
          >
            <path
              d="M15.8856 5L22.8811 12.0046L15.8765 19"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
            <path
              d="M22.4535 11.9591H1.04875"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeWidth="2"
            />
          </mask>
          <g
            mask="url(#mask0_3721_63619)"
            style={
              Object {
                "maskType": "alpha",
              }
            }
          >
            <rect
              fill="currentColor"
              height="24"
              width="24"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithImageSlide 1`] = `
<div>
  <div>
    <ul
      design={
        Object {
          "Title": [Function],
        }
      }
    />
  </div>
  <div />
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithThumbnail 1`] = `
<div>
  <div>
    <ul
      className="scroll-snap-thumb-slider scrollbar-hide"
      design={
        Object {
          "Item": [Function],
        }
      }
    />
  </div>
  <div
    className="flex justify-left"
  >
    <div
      aria-label="Navigation Controls"
      className="flex align-center justify-center opacity-100 controls indicators list-none"
      role="list"
    >
      <div
        className="border-none thumbs indicator block"
        data-index={0}
        tabIndex="0"
        type="button"
      >
        <div
          data-no-hydrate={true}
          id="c978b232fb0971ffe490b7c4ae4e9e44"
          style={
            Object {
              "display": "contents",
            }
          }
        >
          <img
            alt=""
            className="cursor-pointer w-[94px] h-[94px]"
            src="test-file-stub"
            title=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalCarousel Tokens Matches snapshot for WithVerticalThumbs 1`] = `
<div
  className="lg:flex lg:flex-row-reverse"
>
  <div
    className="lg:w-5/6"
  >
    <ul />
  </div>
  <div
    className="pr-16px lg:w-1/6 flex-col"
  >
    <div
      className="flex flex-row justify-between thumbcontrols"
    >
      <div
        className="text-vital-primary-interactive hover:opacity-70 active:text-vital-primary-interactive-active arrow -prev"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="24"
            id="mask0_5803_792"
            maskUnits="userSpaceOnUse"
            style={
              Object {
                "maskType": "alpha",
              }
            }
            width="16"
            x="4"
            y="0"
          >
            <path
              d="M5 8.04436L12.0046 1.04886L19 8.05347"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
            <path
              d="M11.959 1.47632L11.959 22.8811"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeWidth="2"
            />
          </mask>
          <g
            mask="url(#mask0_5803_792)"
            style={
              Object {
                "maskType": "alpha",
              }
            }
          >
            <rect
              fill="currentColor"
              height="24"
              transform="rotate(-90 0 23.9299)"
              width="24"
              y="23.9299"
            />
          </g>
        </svg>
      </div>
      <div
        className="text-vital-primary-interactive hover:opacity-70 active:text-vital-primary-interactive-active arrow -next"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="24"
            id="mask0_5803_806"
            maskUnits="userSpaceOnUse"
            style={
              Object {
                "maskType": "alpha",
              }
            }
            width="16"
            x="4"
            y="0"
          >
            <path
              d="M19 15.8856L11.9954 22.8811L5 15.8765"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
            <path
              d="M12.041 22.4536L12.041 1.04883"
              stroke="#0C0C36"
              strokeLinecap="round"
              strokeWidth="2"
            />
          </mask>
          <g
            mask="url(#mask0_5803_806)"
            style={
              Object {
                "maskType": "alpha",
              }
            }
          >
            <rect
              fill="currentColor"
              height="24"
              transform="rotate(90 24 0)"
              width="24"
              x="24"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
